### DISCUSSION TITLE
How to switch between pages/views?

### DISCUSSION THREAD

**Discussion by markalanboyd** [zauberzeug/nicegui#562] [Q&A] [answered]
I'm looking for a way to create "pages" or "views" that you can navigate through using buttons, tabs, etc. For example: to create a login page that redirects to a home page, which has a button for settings, etc.

In tkinter, you can create frames which hold widgets, and then layer them one on top of another. To navigate through the frame, you simply call that frame to the top of the view and you can then interact with it.

I know this must be possible with NiceGUI, I just can't seem to figure it out from the APIDocs. Any help would be appreciated!

Thank you! :)

**Comment by rodja** [comment, marked as answer]
Have a look at our [modularization demo](https://github.com/zauberzeug/nicegui/tree/main/examples/modularization). Also there is an ongoing discussion in #566 about single-page-applications (SPAs) which you may find helpful.

**Reply by rodja** [reply]
We just merged the SPA example into main: https://github.com/zauberzeug/nicegui/tree/main/examples/single_page_app

**Comment by markalanboyd** [comment]
Thank you so much!
